pkg/checkoutinfo/checkoutinfo.go

```diff
@@ -0,0 +1,229 @@
+package checkoutinfo
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/go-openapi/strfmt"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/buildscript"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model/buildplanner"
+)
+
+type configurer interface {
+	GetBool(string) bool
+}
+
+type projecter interface {
+	Owner() string
+	Name() string
+	BranchName() string
+	LegacyCommitID() string
+	SetLegacyCommit(string) error
+	Dir() string
+	URL() string
+}
+
+type CheckoutInfo struct {
+	auth    *authentication.Auth
+	cfg     configurer
+	project projecter
+}
+
+var ErrBuildscriptNotExist = errors.New("Build script does not exist")
+
+type ErrInvalidCommitID struct {
+	CommitID string
+}
+
+func (e ErrInvalidCommitID) Error() string {
+	return "invalid commit ID"
+}
+
+func New(auth *authentication.Auth, cfg configurer, project projecter) *CheckoutInfo {
+	return &CheckoutInfo{auth, cfg, project}
+}
+
+func (c *CheckoutInfo) Owner() string {
+	return c.project.Owner()
+}
+
+func (c *CheckoutInfo) Name() string {
+	return c.project.Name()
+}
+
+func (c *CheckoutInfo) Branch() string {
+	return c.project.BranchName()
```

Comment:
These should behave the same as the `CommitID()` function; ie. collect from buildscript if opted in.

Reply:
Also their `Set()` methods should be in `checkoutinfo` also.. otherwise we're not syncing the buildscript when those are changed.

Reply:
I'm going to remove the `Owner()`, `Name()`, and `Branch()` accessors since they're not used externally. The ACs make no mention of having checkoutinfo being the gatekeeper of accessors, just for setters like you mentioned here.

If we make checkoutinfo the sole source of truth for Owner, Name, and Branch, then this PR will double in size. I'd rather do that in another ticket.